### PR TITLE
cmd/cel-key: Extract key utility into separate binary to prevent tight coupling between sdk and celestia-node binary

### PR DESCRIPTION
**Pull request by renaynay**
This PR extracts the cosmos sdk key utility from the main celestia-node binary and into a separate `cel-key` binary. 

Use the following flag `--node.type <bridge || full || light>` in order to set the keyring-dir to bridge, light, or full store directory. 

This prevents us having to use cosmos sdk client context initialisation (which is quite messy) inside celestia-node binary.

Thank you @evan-forbes and @Bidon15 for figuring out how to make the key utility actually work after it was broken during the upgrade :) 

### Makefile

`make cel-key` --> builds cel-key binary

`make install-key` --> installs cel-key in GOBIN

### CODE REVIEW

**Review by liamsi — COMMENTED** [review]
This needs @YazzyYaz' attention for docs I assume

File: cmd/celestia/main.go [hunk 1]
```diff
@@ -36,12 +29,17 @@ func main() {
 }
 
 func run() error {
+	cfg := sdk.GetConfig()
+	cfg.SetBech32PrefixForAccount(app.Bech32PrefixAccAddr, app.Bech32PrefixAccPub)
+	cfg.Seal()
+
```
Comment (Wondertan) [inline cmd/celestia/main.go:35]: Why not init? 
Also, It seems weird that this is required for `celestia` binary to function. Can we also extract it into the separate bin or it is required for stateaccess to function?

File: cmd/cel-key/node_types.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func DirectoryFlags() *flag.FlagSet {
+	flags := &flag.FlagSet{}
+	flags.String(nodeDirKey, "", "Sets key utility to use the node type's directory (e.g. "+
+		"~/.celestia-light if --node.type light is passed.")
+
+	return flags
+}
+
+func ParseDirectoryFlags(cmd *cobra.Command) error {
```
Comment (Wondertan) [inline cmd/cel-key/node_types.go:28]: Any chance we can make dir customizable as we extracted this into a separate bin?

**Review by Wondertan — APPROVED** [review]
LGTM and a few questions

File: cmd/cel-key/node_types.go (hunk 2 above)
Comment (renaynay) [reply cmd/cel-key/node_types.go:28]: User can do `--keyring-dir`

File: cmd/celestia/main.go (hunk 1 above)
Comment (renaynay) [reply cmd/celestia/main.go]: good catch

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**